src/components/ListItem.jsx

```diff
@@ -31,8 +47,20 @@ export function ListItem({ name, itemId, purchaseTimestamp }) {
 		setIsPurchased(!isPurchased);
 		if (!isPurchased) {
 			try {
-				await updateItem(listPath, itemId, {
-					dateLastPurchased: new Date(),
+				// name is only passed for testing purposes
```

Comment:
This `handleChange` is a bit overloaded, and could use some reorganization to better encapsulate logical steps in updating an item when it's purchased.

When you take a step back the steps are
1. update all the relevant `Item` properties when a new purchase is recorded
2. update that record in firebase


Right now you're merging the 2 steps together in this handle change. You can instead create a `updateItemOnPurchase` helper that takes in an `item` and all the required inputs to update the relevant fields (`dateLastPurchased`, `dateNextPurchased`, and `totalPurchases`).

This way it's clear now that you have an updated item, you can use the new updated properties to send to firebase


```

// transform step
const updatedItem = updateItemOnPurchase(//inputs);

// update in db step
await updateItem(
  listPath,
  id, {
    dateLastPurchased: updateItem.dateLastPurchased,
    dateNextPurchased: updatedItem.dateNextPurchased,
    totalPurchases: updatedItem.totalPurchases
  }
);


```

Reply:
I am not sure how to rework this part, this is what i came up with:
```
	const updateItemOnPurchase = () => {
		return {
			dateLastPurchased: currentDate,
			dateNextPurchased: calculateDateNextPurchased(currentDate, item),
			totalPurchases: item.totalPurchases + 1,
		}
	}
	const handleChange = async () => {
		setIsPurchased(!isPurchased);
		if (!isPurchased) {
			try {
				const updatedItem = updateItemOnPurchase();
				
				const response = await updateItem(listPath, id, {...updatedItem});
				console.log(response);

			} catch (error) {
				alert(`Item was not marked as purchased`, error.message);
			}
		}
	};
```